### PR TITLE
Feature component disclosure interaction

### LINKED ISSUE #271: Create new disclosure interaction
### Description
Create the component as shown in the video attached below.

### Animation Preview
- Screenshot/Video:

https://github.com/user-attachments/assets/90adb51a-a7af-4189-b364-ed7fc3be2bdc

> [!NOTE]  
> Please refer to the **Animata contributing guidelines** available [here](https://www.animata.design/docs/contributing) for rules on how to contribute to the project. Let us know in the comments if you’re working on this issue, and feel free to ask any questions!

### Requirements
1. Create a new animated component that matches the provided design.
2. Add example usage of the component in the documentation or storybook.

<details>
  <summary>
    <h3> Guidelines & Best Practices</h3>
  </summary>

- Write reusable and clean code following the project’s style guide.
- The component should be **fully customizable**. The user should be able to modify props such as duration, easing, color, and size where applicable.
- Follow the [Contributing Guidelines](https://www.animata.design/docs/contributing) before submitting your pull request.

</details>

> [!IMPORTANT]  
> To ensure more contributors have the opportunity to participate, we kindly request that:
> - Each contributor submits only **one pull request** related to this issue.
> - If you're interested in working on this issue, please comment below. We will assign the issue on a **first-come, first-served basis**.

---

### Additional Resources
Posted on X by: https://x.com/nitishkmrk/status/1833031583739289953

---

> [!Tip]
> Join our Discord community if you need help or have any questions. We're here to help you get started and guide you through the process. [Join now](https://discord.gg/YfvqMf5MTE)

### PR DESCRIPTION
**Pull request by IngaleChinmay04**
The DisclosureInteraction component is an interactive button that expands into a small sheet when clicked. This sheet provides six options (e.g., Project, Task, Note, etc.) and includes a close button to collapse the sheet back to its original state. The component features smooth animations for expanding and collapsing, and is styled using a clean and modern UI design.

Closes #271 

https://github.com/user-attachments/assets/89454af4-7770-478a-897e-93de17d1c737

### CODE REVIEW

File: animata/button/disclosure-interaction.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+          transition: opacity 0.3s ease-in-out;
+        }
+
+        .main-button.hidden {
+          opacity: 0;
+          pointer-events: none;
+        }
+
+        .disclosure-sheet {
+          position: absolute;
```
Comment (hari) [inline animata/button/disclosure-interaction.tsx:79]: Please use tailwindcss.

File: content/docs/button/disclosure-interaction.mdx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+title: Disclosure Interaction
+description: The DisclosureInteraction component is an interactive button that expands into a small sheet when clicked. This sheet provides six options (e.g., Project, Task, Note, etc.) and includes a close button to collapse the sheet back to its original state. The component features smooth animations for expanding and collapsing, and is styled using a clean and modern UI design.
+author: YourTwitterUsername
```
Comment (hari) [inline content/docs/button/disclosure-interaction.mdx:4]: Twitter username is missing

File: content/docs/button/disclosure-interaction.mdx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+```json
+module.exports = {
+  theme: {
+    extend: {
+    }
+  }
+}
+```
+
```
Comment (hari) [inline content/docs/button/disclosure-interaction.mdx:30]: Remove unused steps

File: content/docs/button/disclosure-interaction.mdx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+</Steps>
+
+## Credits
+
+Built by [Your name](https://github.com/YourGithubUsername)
+
+...Add appropriate credits here.
```
Comment (hari) [inline content/docs/button/disclosure-interaction.mdx:53]: Update this

**Review by hari — CHANGES_REQUESTED** [review]
We do not use `npm`, instead we use `yarn`. Also, you cannot downgrade the `next` version. Please remove `package-lock` and update PR.

**Comment by hari** [comment]
Please use 'lucide-react` icons and improve the animation as well :)

**Comment by IngaleChinmay04** [comment]
Okay I will work on the issues

**Comment by IngaleChinmay04** [comment]
I've resolved the requested changes, including Tailwind CSS adjustments, lucide-react icons, and animations. Please review the updates.



https://github.com/user-attachments/assets/dbeec3c6-0de5-487a-b315-8d1c483495d6

**Comment by IngaleChinmay04** [comment]
I mistakenly triggered the preview-deployment for this pull request. Could you please avoid approving the deployment or cancel the approval request?

File: content/docs/button/disclosure-interaction.mdx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+---
+title: Disclosure Interaction
+description: The DisclosureInteraction component is an interactive button that expands into a small sheet when clicked. This sheet provides six options (e.g., Project, Task, Note, etc.) and includes a close button to collapse the sheet back to its original state. The component features smooth animations for expanding and collapsing, and is styled using a clean and modern UI design.
+author: IngaleChinmay04
+---
+
+<ComponentPreview name="button-disclosure-interaction--docs" />
+
+## Usage
+
```
Comment (hari) [inline content/docs/button/disclosure-interaction.mdx:10]: @IngaleChinmay04 you are missing other steps.

File: package.json [hunk 6]
```diff
@@ -46,7 +46,7 @@
     "lodash.startcase": "^4.4.0",
     "lucide-react": "^0.390.0",
     "mdast-util-toc": "^7.1.0",
-    "next": "14.2.10",
+    "next": "13.4",
```
Comment (hari) [inline package.json:49]: @IngaleChinmay04 Please undo this change (and yarn.lock) changes.

**Review by hari — CHANGES_REQUESTED** [review]

**Comment by hari** [comment]
@IngaleChinmay04 The deployment is automatic and it's nothing to worry.

**Comment by IngaleChinmay04** [comment]
I have updated the package.json and added yarn.lock and also steps in mdx file.
Apologies for the inconvenience with these small changes.

File: yarn.lock
Comment (hari) [inline yarn.lock:1]: @IngaleChinmay04 `yarn.lock` is still here. This file shouldn't change as you haven't added any new dependencies.

**Comment by IngaleChinmay04** [comment]
Should I remove the yarn.lock file?

**Comment by IngaleChinmay04** [comment]
I have replcaed the yarn.lock file which was there in main branch to avoid changes

**Review by hari — APPROVED** [review]

**Comment by hari** [comment]
Your PR should not have the `yarn.lock` file. The build is failing.

```
Run yarn install --immutable
➤ YN0000: · Yarn [4](https://github.com/codse/animata/actions/runs/11176462640/job/31069877354?pr=278#step:6:5).2.2
➤ YN0000: ┌ Resolution step
Resolution step
➤ YN0000: └ Completed in 0s 2[8](https://github.com/codse/animata/actions/runs/11176462640/job/31069877354?pr=278#step:6:10)3ms
➤ YN0000: ┌ Post-resolution validation
Post-resolution validation
➤ YN0028: The lockfile would have been modified by this install, which is explicitly forbidden.
➤ YN0000: └ Completed
➤ YN0000: · Failed with errors in 0s 387ms
```

**Comment by hari** [comment]
@IngaleChinmay04 please create a fresh PR with only those 3 files. Make sure to pull the latest main first. I am closing this one.

**Pull request closed without merging**